tests/listobjects/listobjects.go

```diff
@@ -88,6 +109,104 @@ func runTests(t *testing.T, params testParams) {
 	}
 }
 
+func listObjectsAssertion(ctx context.Context, t *testing.T, client tests.ClientInterface, storeID, modelID string, contextTupleTest bool, tuples []*openfgav1.TupleKey, listAssertions []*listobjectstest.Assertion) {
+	for assertionNumber, assertion := range listAssertions {
+		t.Run(fmt.Sprintf("assertion_%d", assertionNumber), func(t *testing.T) {
+			detailedInfo := fmt.Sprintf("ListObject request: %s. Model: %s. Tuples: %s. Contextual tuples: %s", assertion.Request, modelID, tuples, assertion.ContextualTuples)
+
+			ctxTuples := testutils.Shuffle(assertion.ContextualTuples)
+			if contextTupleTest {
+				ctxTuples = append(ctxTuples, tuples...)
+			}
+
+			// assert 1: on regular list objects endpoint
+			resp, err := client.ListObjects(ctx, &openfgav1.ListObjectsRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: modelID,
+				Type:                 assertion.Request.GetType(),
+				Relation:             assertion.Request.GetRelation(),
+				User:                 assertion.Request.GetUser(),
+				ContextualTuples: &openfgav1.ContextualTupleKeys{
+					TupleKeys: ctxTuples,
+				},
+				Context: assertion.Context,
+			})
+
+			if assertion.ErrorCode == 0 {
+				require.NoError(t, err, detailedInfo)
+				require.ElementsMatch(t, assertion.Expectation, resp.GetObjects(), detailedInfo)
+			} else {
+				require.Error(t, err, detailedInfo)
+				e, ok := status.FromError(err)
+				require.True(t, ok, detailedInfo)
+				require.Equal(t, assertion.ErrorCode, int(e.Code()), detailedInfo)
+			}
+
+			// assert 2: on streaming list objects endpoint
+			done := make(chan struct{})
+			var streamedObjectIDs []string
+
+			clientStream, err := client.StreamedListObjects(ctx, &openfgav1.StreamedListObjectsRequest{
+				StoreId:              storeID,
+				AuthorizationModelId: modelID,
+				Type:                 assertion.Request.GetType(),
+				Relation:             assertion.Request.GetRelation(),
+				User:                 assertion.Request.GetUser(),
+				ContextualTuples: &openfgav1.ContextualTupleKeys{
+					TupleKeys: ctxTuples,
+				},
+				Context: assertion.Context,
+			}, []grpc.CallOption{}...)
+			require.NoError(t, err)
+
+			var streamingErr error
+			var streamingResp *openfgav1.StreamedListObjectsResponse
+			go func() {
+				for {
+					streamingResp, streamingErr = clientStream.Recv()
+					if streamingErr == nil {
```

Comment:
Stick to the normal go pattern of `streamingErr != nil` and handle the exit there. Also, seems this could swallow errors, do we want this? Perhaps consider using a a workgroup.

Reply:
🙇🏽 

If you can also update the other code, would be great.

Reply:
```
if streamingErr != nil {
     if errors.Is(streamingErr, io.EOF) {
        break
      }
      return streamingErr
}
streamedObjectIDs = append(streamedObjectIDs, streamingResp.GetObject())
```